.github/workflows/ci-test.yml

```diff
@@ -0,0 +1,143 @@
+      test-rust: ${{ steps.rust-changes.outputs.changes-found }}
+      test-dev-setup: ${{ steps.dev-setup-sh-changes.outputs.changes-found }}
+      test-website-build: ${{ steps.website-changes.outputs.changes-found }}
+      test-non-rust-lint: ${{ steps.non-rust-lint-changes.outputs.changes-found }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0 #get all the history!!!
+      - id: changes
+        name: determine changes
+        uses: ./.github/actions/changes
+        with:
+          workflow-file: ci-test.yml
+      - id: rust-changes
+        name: find rust/cargo changes.
+        uses: ./.github/actions/matches
+        with:
+          pattern: '**.rs\|Cargo.toml\|Cargo.lock'
+      - id: non-rust-lint-changes
+        name: find shell/dockerfile changes
+        uses: ./.github/actions/matches
+        with:
+          pattern: '**.Dockerfile\|**.sh'
+      - id: dev-setup-sh-changes
+        name: find dev-setup.sh/base docker image changes
+        uses: ./.github/actions/matches
+        with:
+          pattern: '.github/workflows/ci-publish-base-image.yml\|docker/ci/**/Dockerfile\|scripts/dev_setup.sh'
+      - id: website-changes
+        name: find website changes.
+        uses: ./.github/actions/matches
+        with:
+          pattern: "documentation/**"
+
+  dev-setup-sh-test:
+    runs-on: ubuntu-latest-xl
+    continue-on-error: false
+    needs: prepare
+    if: ${{ needs.prepare.outputs.test-dev-setup == 'true' }}
+    strategy:
+      matrix:
+        target_os: [alpine, arch, centos, github]
+    steps:
+      - uses: actions/checkout@v2
+      - name: build image with dev-setup.sh
+        run: docker build -f docker/ci/${{ matrix.target_os }}/Dockerfile -t libra/build_environment:test .
+
+  non-rust-lint:
+    runs-on: ubuntu-latest
+    continue-on-error: false
+    needs: prepare
+    if: ${{ needs.prepare.outputs.test-non-rust-lint == 'true'  }}
+    container:
+      image: libra/build_environment:github-1
+    steps:
+      - uses: actions/checkout@v2
+      - name: shell lints
+        run: |
+          shellcheck scripts/dev_setup.sh && \
+          shellcheck scripts/dockerhub_prune.sh && \
+          shellcheck .github/actions/changes/get_pr_info.sh && \
+          shellcheck docker/build_push.sh && \
+          shellcheck docker/dockerhub_to_novi_ecr.sh
+      - name: docker lints
+        run: |
+          hadolint docker/ci/github/Dockerfile && \
+          hadolint docker/ci/arch/Dockerfile && \
+          hadolint docker/ci/alpine/Dockerfile && \
+          hadolint docker/ci/centos/Dockerfile
+
+  unit-test:
+    runs-on: ubuntu-latest-xl
+    continue-on-error: false
+    needs: prepare
+    #if: ${{ needs.prepare.outputs.test-rust == 'true' }}
+    container:
+      image: libra/build_environment:github-1
+    env:
+      RUST_BACKTRACE: 1
+    steps:
+      - name: stats
```

Comment:
okay past this is a mess and needs refactoring, but it's passing right now with move tools config out of the way.   see notes about build locations in the github action/home folders and dev's ~/bin/ dir.    Need to fix this to get full working tests.   Not sure about sccache yet.